### PR TITLE
[Container Registry] Acr delete repo

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_base_client.py [hunk 1]
```diff
@@ -32,8 +31,7 @@ class ContainerRegistryBaseClient(object):
 
     def __init__(
         self, endpoint: str, credential: "AsyncTokenCredential", **kwargs
-    ):  # pylint:disable=client-method-missing-type-annotations
-        # type: (str, TokenCredential, Dict[str, Any]) -> None
```
Comment (swathipil) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_base_client.py:36]: just curious, but why did you remove this line?

Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_base_client.py:36]: This is async, so always Python3, we can add the type hints directly into the signature.

**Review by iscai-msft — APPROVED** [review]

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_base_client.py [hunk 2]
```diff
@@ -32,8 +31,7 @@ class ContainerRegistryBaseClient(object):
...
+    ):  # pylint: disable=client-method-missing-type-annotations
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_base_client.py:34]: seems like we could delete this pylint ignore. Not sure if it's yelling about the return type, though

File: sdk/containerregistry/azure-containerregistry/tests/testcase.py [hunk 3]
```diff
@@ -101,12 +104,66 @@ def get_token(self, *args):
...
     def __init__(self, method_name):
         super(ContainerRegistryTestClass, self).__init__(method_name)
         self.vcr.match_on = ["path", "method", "query"]
         self.recording_processors.append(AcrBodyReplacer())
 
+    def sleep(self, t):
+        if self.is_live:
+            time.sleep(t)
+
+    def _fake_sleep(self, *args, **kwargs):
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/tests/testcase.py:116]: is this used anywhere or do you have future plans for it?

**Review by kristapratico — APPROVED** [review]

File: sdk/containerregistry/azure-containerregistry/tests/testcase.py (hunk 3 above)
Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/tests/testcase.py:149]: Nope, I had plans to use it but I'm going to remove it for now.
